content/PreparedStatement/PreparedStatement.q.md

```diff
@@ -0,0 +1,1708 @@
+        if ($conn) {
+            echo "Successfully connected to the database. <br />";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once("connection.php");
+
+    $pstmt = $conn->prepare("SELECT * FROM students WHERE gender = ?");
+    $pstmt->execute(array('Female'));
+    $data = $pstmt->fetchAll();
+    
+    foreach ($data as $row) {
+        echo $row['student_id'].$row['first_name'].$row['last_name'].$row['birth_date'].$row['gender']. ' ';
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `2SamanthaDanes1999-10-12Female 4VikkiFernando2005-05-17Female `.
+
+- It prints `Successfully connected to the database.` and `2SamanthaDanes1999-10-12Female 4VikkiFernando2005-05-17Female 6AlisaElls1999-06-30Female `.
+
+
+/// type=MS, answer=[2,3,4,5]
+
+Which statements correctly describe the code on line 4 of `student.php`?
+
+- It queries the SQL statement.
+
+- It returns the `PDOStatement` object.
+
+- It calls the `prepare()` method of the `$conn` object.
+
+- It assigns the `PDOStatement` object to the variable `$pstmt`.
+
+- It prepares the argument `SELECT * FROM students WHERE gender = ?` in the `prepare()` method.
+
+
+/// type=MS, answer=[1,5]
+
+Which statements correctly describe `WHERE gender = ?` on line 4 of `student.php`?
+
+- It returns all records of students whose gender is equal to `Female`.
+
+- It deletes all columns in the `students` table whose gender is equal to `?`.
+
+- It returns all records of students in the `students` table whose gender is equal to `?`.
+
+- It updates all columns in the `students` table which contains a question mark `?` as gender.
+
+- It uses the positional placeholder `?` to associate the value of `gender` to be filtered in the `students` table.
+
+
+/// type=MS, answer=[1,3,5]
+
+Which statements correctly describe `$pstmt->execute(array('Female'))` on line 5 of `student.php`?
+
+- It returns the boolean value `true`.
+
+- It holds the value `Female` in the SQL statement.
+
+- It calls the `execute()` method of the `$pstmt` object.
+
+- It assigns the specified value `Female` to the SQL statement.
+
+- It executes the argument `array('Female')` in the `execute()` method.
+
+
+/// type=SS, answer=[5]
+
+In `student.php`, what does `$data` do on line 6?
+
+- It holds the SQL statement.
+
+- It executes the SQL statement.
+
+- It stores the prepared statement.
+
+- It connects to the PostgreSQL database.
+
+- It holds the array of rows in the result set.
+
+
+/// type=MS, answer=[2,3,5]
+
+Which statements correctly describe the code on line 6 of `student.php`?
+
+- It queries the SQL statement.
+
+- It assigns the returned value to the variable `$data`.
```

Comment:
What returned value is assigned to `$data`? and from where? Please check. Thanks.